.github/actions/setup-binary-builds/action.yml

```diff
@@ -109,16 +109,28 @@ runs:
           set -euxo pipefail
           CONDA_ENV="${RUNNER_TEMP}/conda_environment_${GITHUB_RUN_ID}"
 
-          conda create \
-            --yes --quiet \
-            --prefix "${CONDA_ENV}" \
-            "python=${PYTHON_VERSION}" \
-            cmake=3.22 \
-            conda-build=3.27.0 \
-            conda=23.10.0 \
-            ninja=1.10 \
-            pkg-config=0.29 \
-            wheel=0.37
+          if [[ "${PYTHON_VERSION}" = "3.12" ]]; then
+            conda create \
+              --yes --quiet \
+              --prefix "${CONDA_ENV}" \
+              "python=3.9" \
+              cmake=3.22 \
+              conda=23.10.0 \
+              ninja=1.10 \
+              pkg-config=0.29 \
+              wheel=0.37
+          else
+            conda create \
+              --yes --quiet \
+              --prefix "${CONDA_ENV}" \
+              "python=${PYTHON_VERSION}" \
+              cmake=3.22 \
+              conda-build=3.27.0 \
+              conda=23.10.0 \
+              ninja=1.10 \
+              pkg-config=0.29 \
+              wheel=0.37
+          fi
```

Comment:
```suggestion
          if [[ "${PYTHON_VERSION}" = "3.12" ]]; then
              TODO: Remove me no later than Dec 15th 2023
              echo "Python-3.12 conda builds are not ready yet"
              PYTHON_VERSION = 3.9
          fi

          conda create \
            --yes --quiet \
            --prefix "${CONDA_ENV}" \
            "python=${PYTHON_VERSION}" \
            cmake=3.22 \
            conda-build=3.27.0 \
            conda=23.10.0 \
            ninja=1.10 \
            pkg-config=0.29 \
            wheel=0.37
```

Reply:
This won't work since wheel builds are using the same setup script. The only difference conda-build is not required for wheel builds. We build wheels within conda env.
Please take a look: https://github.com/pytorch/test-infra/blob/main/.github/workflows/build_wheels_linux.yml#L180